hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegionServer.java

```diff
@@ -530,8 +530,8 @@ public HRegionServer(final Configuration conf) throws IOException {
 
       regionServerAccounting = new RegionServerAccounting(conf);
 
-      blockCache = BlockCacheFactory.createBlockCache(conf);
       DataTieringManager.instantiate(onlineRegions);
+      blockCache = BlockCacheFactory.createBlockCache(conf);
```

Comment:
Instantiation of Data tiering manager is required to read the cache from persistence. Hence, change the order here.